examples/derive_macro.rs

```diff
@@ -0,0 +1,104 @@
+#[cfg(feature = "derive")]
+use redb::{Database, TableDefinition, Value};
+
+#[cfg(feature = "derive")]
+use redb_derive::Value as DeriveValue;
+
+#[cfg(feature = "derive")]
+use tempfile::NamedTempFile;
+
+#[cfg(feature = "derive")]
+#[derive(Debug, DeriveValue)]
+#[redb(type_name = "User")]
```

Comment:
I changed my mind about this `#[redb(type_name)]` annotation. Let's remove it and just have the macro derive the name using the name of the struct